common/json_writer/src/lib.rs

```diff
@@ -0,0 +1,53 @@
+use std::fmt::Write;
+
+pub struct JsonWriter {
+    buffer: String,
+}
```

Comment:
Having this struct is really good idea. However, the unit tests are missing.

I also wonder about having no error checking and all the method public.
* I think it would be better to have few high level public methods (say `write_key_value`, `write_key_open_obj`, `write_close_obj`).

Reply:
I feel fine granular functions give more control over the structure. For example, write_key_value is different on different keys. For example time and type are strings and different in the case of c8y and thin edge serialization.